### PR TITLE
Heitor's updates to terrame 2.0-beta-3 - part 2

### PR DESCRIPTION
**Pull request by hguerra**
- [x] #1257
- [x] #1262
- [x] #1267 
- [x] #1284

### CODE REVIEW

File: packages/base/lua/ErrorHandling.lua [hunk 1]
```diff
@@ -554,6 +554,25 @@ function suggestion(value, options)
 	end
 end
 
+--- Return the arguments of suggestion within a question " Do you mean "..suggestion.."'?"
```
Comment (pedro-andrade-inpe) [inline packages/base/lua/ErrorHandling.lua]: "Do you mean (missing ') [...]" in the documentation.

File: packages/base/tests/functional/basics/ErrorHandling.lua [hunk 2]
```diff
@@ -237,6 +237,16 @@ return{
...
 	end,
+	suggestionMsg = function(unitTest)
+		local t = {
+			aaaaa = true,
+			bbbbb = true,
+			ccccc = true
+		}
+
+		unitTest:assertEquals(suggestionMsg(suggestion("aaaab", t)), " Do you mean 'aaaaa'?")
+		unitTest:assertEquals(suggestionMsg(nil), "")
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/ErrorHandling.lua]: Do not use nil as an argument for a function. Just call it without any argument.

File: packages/base/tests/functional/basics/UnitTest.lua [hunk 3]
```diff
@@ -55,8 +55,8 @@ return{
 		unitTest:assertEquals(1, 1.1, 0.15)
 		unitTest:assertEquals("abc", "abd", 1)
 
-		local expected = [[string [biomassa-manaus.asc] ]]
 		local actual = [[string [/home/jenkins/Documents/ba1c13592dcf65f3d0b2929f8eff266c4e622470/install/bin/packages/terralib/data/biomassa-manaus.asc] ]]
+		local expected = [[string [biomassa-manaus.asc] ]]
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/UnitTest.lua]: It is better to use `"` to describe strings that have only one line. Use `[[ ]]` only for strings with multiple lines.

File: packages/base/doc/index.html [hunk 4]
```diff
@@ -777,7 +777,7 @@
 						<b>License: </b>LGPL-3<br>
 						
 
-						<b>Release: </b>18 July 2016<br>
+						<b>Release: </b>20 July 2016<br>
```
Comment (pedro-andrade-inpe) [inline packages/base/doc/index.html]: It is better to update html files in a separate commit.

File: packages/base/tests/functional/basics/CellularSpace.lua [hunk 5]
```diff
@@ -175,7 +175,98 @@ return{
...
+
+		cs = CellularSpace{
+			project = projName,
+			layer = clName1,
+			geometry = true
+		}
+
+		forEachCell(cs, function(c)
+			unitTest:assertNotNil(c.geom)
+			unitTest:assertNil(c.OGR_GEOMETRY)
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/CellularSpace.lua]: Check the attribute names here and their types.

File: packages/base/lua/CellularSpace.lua [hunk 6]
```diff
@@ -396,6 +402,24 @@ local function setCellsByTerraLibDataSet(self, dSet)
 	end
 end
 
+local function loadGdal(self)
+	local tlib = terralib.TerraLib{}
+	local dSet = tlib:getGdalByFilePath(self.file)
+	setCellsByTerraLibDataSet(self, dSet)
+	local temp = ""
+
+	for i = self.file:len(), 1, -1 do
```
Comment (hguerra) [inline packages/base/lua/CellularSpace.lua]: Utils:getFileNameWithExtension

File: packages/base/tests/functional/basics/CellularSpace.lua [hunk 7]
```diff
@@ -267,6 +277,168 @@ return{
...
+
+		unitTest:assertEquals(projName, cs.project.file)
+		unitTest:assertType(cs.layer, "Layer")
+
+		unitTest:assertEquals(proj.title, title)
+		unitTest:assertEquals(proj.author, author)
+
+		unitTest:assertEquals(layer.source, "tif")
+		unitTest:assertEquals(layer.file, filePath1)
+		unitTest:assert(#cs.cells > 0)
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/CellularSpace.lua]: Use assertEquals here, with the exact number of pixels in the data.

File: packages/base/tests/functional/basics/CellularSpace.lua [hunk 8]
```diff
@@ -267,6 +277,168 @@ return{
...
+		layer = terralib.Layer{
+			project = proj,
+			name = layerName1,
+			file = filePath1
+		}
+
+		cs = CellularSpace{
+			project = projName,
+			layer = layerName1
+		}
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/CellularSpace.lua]: How can one get the values of the pixels of the tif file?

**Comment by pedro-andrade-inpe** [comment]
Bad news, base package tests aborted TerraME:

```
Testing tests/functional/basics/CellularSpace.lua
Testing add
Testing CellularSpace
/home/jenkins/Configs/terrame/status/request.sh: line 80: 21795 Segmentation fault      (core dumped) terrame -color $TERRAME_PACKAGE $TERRAME_RUN_TYPE config.lua 2> /dev/null
Sending build status to GitHub
Removing /home/jenkins/Documents/f8eef72f69a0157b53c5f14d548fdbc5d9e13391
Build step 'Conditional steps (multiple)' marked build as failure
Xvfb stopping
Finished: FAILURE
```

**Pull request merged**